ui/src/api/googleOauth.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  API_CLIENT_ID,
+  API_CLIENT_SECRET,
+  GOOGLE_CLIENT_ID,
+  REDIRECT_URL,
+  SERVER_URL,
+} from '../globals/constants';
+import {setToken} from '../utils/token';
+import {NavigateFunction} from 'react-router-dom';
+
+export const googleOauth = async () => {
+  window.location.href =
+    'https://accounts.google.com/o/oauth2/v2/auth?' +
+    'scope=https%3A//www.googleapis.com/auth/userinfo.email%20' +
+    'https%3A//www.googleapis.com/auth/userinfo.profile&' +
+    'include_granted_scopes=true&' +
+    'response_type=token&' +
+    'state=state_parameter_passthrough_value&' +
+    `redirect_uri=${REDIRECT_URL}&` +
+    `client_id=${GOOGLE_CLIENT_ID}`;
```

Comment:
[URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) can be used to make it more readable. It will encode the parameters as well.:
```ts
const queryParams = new URLSearchParams({
  scope: 'https://www.googleapis.com/auth/userinfo.email,https://www.googleapis.com/auth/userinfo.profile'
...
});

window.location.href = `https://accounts.google.com/o/oauth2/v2/auth?${queryParams.toString()}`;
```